velox/exec/Driver.cpp

```diff
@@ -355,15 +355,31 @@ void Driver::pushdownFilters(int operatorIndex) {
   op->clearDynamicFilters();
 }
 
-RowVectorPtr Driver::next(std::shared_ptr<BlockingState>& blockingState) {
+RowVectorPtr Driver::next(ContinueFuture* future) {
   enqueueInternal();
   auto self = shared_from_this();
   facebook::velox::process::ScopedThreadDebugInfo scopedInfo(
       self->driverCtx()->threadDebugInfo);
   ScopedDriverThreadContext scopedDriverThreadContext(*self->driverCtx());
+  std::shared_ptr<BlockingState> blockingState;
   RowVectorPtr result;
   auto stop = runInternal(self, blockingState, result);
 
+  if (blockingState) {
+    VELOX_DCHECK_NULL(result)
+    *future = blockingState->future();
+    return nullptr;
+  }
+
+  if (stop == StopReason::kPause) {
+    VELOX_DCHECK_NULL(blockingState)
+    VELOX_DCHECK_NULL(result)
+    ContinueFuture resumeFuture;
+    task()->pauseRequested(&resumeFuture);
+    *future = std::move(resumeFuture);
+    return nullptr;
+  }
```

Comment:
This makes single-threaded execution work correctly with pause API which is called during arbitration.